tests/plugins/dialog/focus.js

```diff
@@ -0,0 +1,319 @@
+					tab: 'mp-test1',
+					elementId: 'mp-input11'
+				} ) )
+				.then( focusElement( { direction: 'previous' } ) )
+				.then( assertFocusedTab( 'mp-test1' ) )
+				.then( focusElement( { key: KEYS.ARROW_RIGHT } ) )
+				.then( assertFocusedTab( 'mp-test2' ) )
+				.then( focusElement( { key: KEYS.ARROW_UP } ) )
+				.then( assertFocusedTab( 'mp-test1' ) )
+				.then( focusElement( { key: KEYS.ARROW_DOWN } ) )
+				.then( focusElement( { key: KEYS.ARROW_DOWN } ) )
+				.then( assertFocusedTab( 'mp-test3' ) )
+				.then( focusElement( { key: KEYS.ARROW_LEFT } ) )
+				.then( assertFocusedTab( 'mp-test2' ) );
+		},
+
+		'test multi page dialog should bring the focus to the tab with the ALT+F10 keys': function() {
+			var bot = this.editorBot;
+
+			return bot.asyncDialog( 'multiPageDialog' )
+				.then( ie11fix )
+				.then( assertFocusedElement( {
+					tab: 'mp-test1',
+					elementId: 'mp-input11'
+				} ) )
+				.then( focusElement( { direction: 'next' } ) )
+				.then( focusElement( { direction: 'next' } ) )
+				.then( assertFocusedElement( {
+					tab: 'mp-test1',
+					elementId: 'mp-input13'
+				} ) )
+				.then( focusElement( { key: KEYS.F10, altKey: true } ) )
+				.then( assertFocusedTab( 'mp-test1' ) )
+				.then( focusElement( { direction: 'previous' } ) )
+				.then( focusElement( { direction: 'previous' } ) )
+				.then( assertFocusedElement( {
+					buttonName: 'cancel'
+				} ) )
+				.then( focusElement( { key: KEYS.F10, altKey: true } ) )
+				.then( assertFocusedTab( 'mp-test1' ) );
+		},
+
+		'test hidden page dialog should skip the focus of the hidden element on the page': function() {
+			var bot = this.editorBot;
+
+			return bot.asyncDialog( 'hiddenPageDialog' )
+				.then( ie11fix )
+				.then( assertFocusedElement( {
+					tab: 'hp-test1',
+					elementId: 'hp-input11'
+				} ) )
+				.then( focusElement( { direction: 'previous' } ) )
+				.then( focusElement( { key: KEYS.ARROW_RIGHT } ) )
+				.then( assertFocusedTab( 'hp-test2' ) )
+				.then( focusElement( { direction: 'next' } ) )
+				.then( assertFocusedElement( {
+					tab: 'hp-test2',
+					elementId: 'hp-input22'
+				} ) )
+				.then( focusElement( { direction: 'next' } ) )
+				.then( assertFocusedElement( {
+					buttonName: 'cancel'
+				} ) );
+		},
+
+		'test hidden page dialog should skip the focus from the hidden tab on the page': function() {
+			var bot = this.editorBot;
+
+			return bot.asyncDialog( 'hiddenPageDialog' )
+				.then( ie11fix )
+				.then( assertFocusedElement( {
+					tab: 'hp-test1',
+					elementId: 'hp-input11'
+				} ) )
+				.then( focusElement( { direction: 'previous' } ) )
+				.then( focusElement( { key: KEYS.ARROW_RIGHT } ) )
+				.then( assertFocusedTab( 'hp-test2' ) )
+				.then( focusElement( { key: KEYS.ARROW_RIGHT } ) )
+				.then( assertFocusedTab( 'hp-test4' ) );
+		}
+	};
+
+	tests = bender.tools.createAsyncTests( tests );
+
+	CKEDITOR.tools.extend( tests, {
+		init: function() {
```

Comment:
I think this should also be a part `dialogTools`. Since dialog definitions are useless without initiating dialogs it's the way it will be used in 95% of cases so we can benefit from making it a common function.